frontend/catalyst/device/qjit_device.py

```diff
@@ -440,6 +440,17 @@ def _measurement_transform_program(self):
             else:
                 raise RuntimeError("The device does not support observables or sample/counts")
 
+        elif self.measurement_processes in [{"Sample"}, {"Counts", "Sample"}, {"Counts"}]:
+            # ToDo: this branch should become unneccessary when selective conversion of unsupported MPs is finished, see ToDo below
+            if not split_non_commuting in measurement_program:
+                measurement_program.add_transform(split_non_commuting)
+            mp_transform = (
```

Comment:
@lillian542 Not sure that the tests trigger those lines?

Reply:
The relevant test was indented and didn't run 🙈 Should be good now.